mcp-spring/mcp-spring-webflux/src/main/java/io/modelcontextprotocol/server/transport/WebFluxSseServerTransportProvider.java

```diff
@@ -314,7 +314,7 @@ private Mono<ServerResponse> handleMessage(ServerRequest request) {
 		return request.bodyToMono(String.class).flatMap(body -> {
 			try {
 				McpSchema.JSONRPCMessage message = McpSchema.deserializeJsonRpcMessage(objectMapper, body);
-				return session.handle(message).flatMap(response -> ServerResponse.ok().build()).onErrorResume(error -> {
+				return session.handle(message).then(Mono.defer(() -> ServerResponse.ok().build())).onErrorResume(error -> {
```

Comment:
Why do you need to call the Mono.defer()?  
Since the ServerResponse.ok().build() returns a Mono and the execution is lazy.

Mono.defer() creates a runtime wrapper as unnecessary overhead.

Reply:
hum, yes, the defer is nuneeded